examples/mnist_3d/analysis.jl

```diff
@@ -0,0 +1,105 @@
+using BSON
+using CUDA
+using Flux
+using Images
+using ProgressMeter
+using Plots
+using Random
+using Statistics
+using TOML
+
+using CliMAgen
+
+package_dir = pkgdir(CliMAgen)
+include(joinpath(package_dir, "examples/utils_data.jl"))
+include(joinpath(package_dir, "examples/utils_analysis.jl"))
+
+function run_analysis(params; FT=Float32, logger=nothing)
+    # unpack params
+    savedir = params.experiment.savedir
+    rngseed = params.experiment.rngseed
+    nogpu = params.experiment.nogpu
+
+    inchannels = params.model.inchannels
+
+    nsamples = params.sampling.nsamples
+    nsteps = params.sampling.nsteps
+    sampler = params.sampling.sampler
+
+    # set up rng
+    rngseed > 0 && Random.seed!(rngseed)
+
+    # set up device
+    if !nogpu && CUDA.has_cuda()
+        dev = Flux.gpu
+        @info "Sampling on GPU"
+    else
+        dev = Flux.cpu
+        @info "Sampling on CPU"
+    end
+
+    # set up dataset
+    dl, _ = get_data_mnist_3d(
+        nsamples;
+        FT=FT
+    )
+    xtrain, dl = Iterators.peel(dl)
+
+    # set up model
+<<<<<<< HEAD
+    checkpoint_path = joinpath(savedir, "checkpoint_250.0.bson")
+=======
+    checkpoint_path = joinpath(savedir, "checkpoint.bson")
+>>>>>>> a55a146 (working mnist_3d, working tests)
+    BSON.@load checkpoint_path model model_smooth opt opt_smooth
+    model = dev(model)
+
+    # sample from the trained model
+    t = ones(FT, nsamples) |> dev
+    init_z = randn(FT, (64, 64, 16, 1, nsamples)) |> dev
+    _, σ_T = CliMAgen.marginal_prob(model, zero(init_z), t)
+    init_x = (σ_T .* init_z)
+    time_steps = LinRange(1.0f0, 1.0f-3, nsteps)
+    Δt = time_steps[1] - time_steps[2]
+    samples = Euler_Maruyama_sampler(model, init_x, time_steps, Δt, nspatial=3)
+    samples = cpu(samples)
+
+    # loss curves
+    loss_plot(savedir, "losses.png"; xlog = false, ylog = true)    
+
+    # make some gifs
+    nspatial = 3
+    for i in 1:nsamples
+        # xtrain    
+        anim = @animate for j ∈ 1:size(xtrain, nspatial)
+            heatmap(xtrain[:,:,j,1,i])
```

Comment:
we only have one channel here, but is it worth generalizing already to 
for ch in 1:inchannels
....heatmap(xtrain[:,:,j,ch,i])
end
?

Reply:
Yea, if we move this into a function we should do this, but this code is very slow, so before that, I would like to make it faster :(.